js/src/utils/external.ts

```diff
@@ -0,0 +1,77 @@
+
+import { spawn } from "child_process";
+import { IS_DEVELOPMENT_OR_CI, TELEMETRY_URL } from "../sdk/utils/constants";
+import { ifObjectStringify } from "../sdk/utils/common";
+
+/**
+ * Sends a reporting payload to the telemetry server using a child process.
+ * This function is intended for use in Node.js environments.
+ * 
+ * @param {any} reportingPayload - The payload to be sent to the telemetry server.
+ */
+export async function sendProcessReq(info:{
+    url: string,
+    method: string,
+    headers: Record<string, string>,
+    data: Record<string, unknown>
+}) {
+     if(IS_DEVELOPMENT_OR_CI){
+        console.log(`Hitting ${info.url}[${info.method}] with ${ifObjectStringify(info.data)}`);
+        return true;
+    }
+    try {
+        // Spawn a child process to execute a Node.js script
+        const child = spawn('node', ['-e', `
+        const axios = require('axios');
```

Comment:
the function attempts to require the axios library in the child process without ensuring its availability, which can lead to errors that are silently ignored, preventing telemetry data from being sent.